amortization.py

```diff
@@ -231,8 +254,38 @@ def __str__(self):
                    (self.rate, self.nper, self.pv, self.typ))
 
 
+    def _calc_pmt_diff_(self, pmt):
+        """ Difference between last payment and penultimate payment """
+        l = Loan(self.rate, self.nper, self.pv, self.date, payment=pmt)
+        return l.period(l.nper).payment - l.period(l.nper-1).payment
+
     def pmt(self):
-        return self.payment or pmt(self.rate, self.nper, self.pv, self.typ)
+        if self.payment:
+            return self.payment
+
+        if PMT_METHOD == ORDINARY_ANNUITY_PMT_METHOD:
```

Comment:
Can you please explain what the newton's method approach is doing? I see that it's iterating using what looks like a finite difference method until the last two payments are close enough but I am confused by the creation of loans with the payment as an input and then using payments again to calculate the difference.